lib/model/QubitRepository.php

```diff
@@ -124,24 +124,45 @@ public function save($connection = null)
     return $this;
   }
 
+  public function getRelatedInformationObjectIds()
+  {
+    $sql = "SELECT id FROM ". QubitInformationObject::TABLE_NAME ." WHERE repository_id=:repository_id";
+
+    $params = array(':repository_id' => $this->id);
+    $results = QubitPdo::fetchAll($sql, $params, array('fetchMode' => PDO::FETCH_ASSOC));
+
+    $ioIds = array();
+
+    foreach ($results as $row)
+    {
+      array_push($ioIds, $row['id']);
+    }
+
+    return $ioIds;
+  }
+
   public function updateRelatedIos()
   {
     $ioIds = array();
     $context = sfContext::getInstance();
     $env = $context->getConfiguration()->getEnvironment();
 
-    foreach ($this->informationObjects as $io)
+    foreach ($this->getRelatedInformationObjectIds() as $id)
     {
       // Update synchronously in CLI tasks and jobs
       if (in_array($env, array('cli', 'worker')))
       {
+        $io = QubitInformationObject::getById($id);
         QubitSearch::getInstance()->update($io, array('updateDescendants' => true));
       }
       else
       {
         // Otherwise save id to update asynchronously
-        $ioIds[] = $io->id;
+        $ioIds[] = $id;
```

Comment:
As @sbreker said, this is really unnecessary. I introduced this logic, but it's far from ideal ...  🤦

I'd suggest to check the environment first and trigger the job directly as you already have the ids array. In pseudocode ...

```
if not related io ids
{
  return
}

if not in cli and not in worker
{
  launch job
  set flash
  return
}

foreach related io ids
{
  get io
  update
  clear cache
}
```

Reply:
Nice!